Lib/unittest/mock.py

```diff
@@ -431,6 +452,11 @@ def _mock_add_spec(self, spec, spec_set, _spec_as_instance=False,
                        _eat_self=False):
         _spec_class = None
         _spec_signature = None
+        _spec_asyncs = []
+
+        for attr in dir(spec):
+            if asyncio.iscoroutinefunction(getattr(spec, attr, None)):
```

Comment:
Late note, not sure if anyone gets a notification for this: This broke a use case for me. Before this change, the `Mock` constructor did not call `getattr` on any of the attributes of the mocked object specification. I used it to mock an instance of a base class that does not have all its properties implemented, using the `**kwargs` in the `Mock` constructor to ensure the unimplemented properties got a sane value.
I can solve this by monkey-patching the base class instance *before* passing it to `Mock()`. But maybe it would be wise to skip all attributes here that appear in `**kwargs`, because they will be subsequently overwritten?

Reply:
> not sure if anyone gets a notification for this

For this reason it's worth creating a new issue and mentioning this PR there.

Reply:
@f0k - better yet, if you could work up a PR with a test that shows the problem you encountered, along with a fix for it, that would be better!

Reply:
@f0k - better yet, if you could work up a PR with a test that shows the problem you encountered, along with a fix for it, that would be better!

Reply:
@fok some related issues . Please use search before creating the issue so that you can check for previous issues and make sure it's not duplicate. Thanks.

https://github.com/python/cpython/issues/89917
https://github.com/python/cpython/issues/85934

Reply:
@tirkarthi you probably meant @f0k, not me

Reply:
> For this reason it's worth creating a new issue

I wasn't sure yet whether this warrants a fix, and was aiming low :)

> some related issues

Good spot, yes, they're both the same problem, and there are already three PRs proposing a fix. I will continue the discussion in #85934, as it's the older one of the two issues.